test/spec/znkTimeline/znTimelineDrv.spec.js

```diff
@@ -0,0 +1,118 @@
+        }
+    ]));
+
+    function createDirectiveHtml($scope, content, onFinishCb) {
+        if (!$scope) {
+
+            $scope = $rootScope.$new();
+            $scope.currentSubjectId = 2;
+
+            $scope.timeLineData = {
+                data:[
+                    { exerciseId: 1105,
+                        exerciseType: 4,
+                        score: 90,
+                        time: 1444503875221
+                    },
+                    { exerciseId: 1105,
+                        exerciseType: 1,
+                        score: 190,
+                        time: 1444503875221
+                    },
+                    { exerciseId: 1105,
+                        exerciseType: 4,
+                        score: 20,
+                        time: 1444503875221
+                    },
+                    { exerciseId: 1105,
+                        exerciseType: 3,
+                        score: 10,
+                        time: 1444503875221
+                    }
+                ],
+                id:  $scope.currentSubjectId
+            };
+
+
+            for(var i=0 ;i <4; i ++){
+                $scope.timeLineData.data[4+i] = $scope.timeLineData.data[i];
+                $scope.timeLineData.data[8+i] = $scope.timeLineData.data[i];
+                $scope.timeLineData.data[12+i] = $scope.timeLineData.data[i];
+            }
+
+            var optionsPerDevice = {
+                height: 150,
+                distance: 90,
+                upOrDown: 75,
+                yUp: 65,
+                yDown: 75
+            };
+
+            $scope.options = {
+                colors: ['#75cbe8', '#f9d41b', '#ff5895', '', '','#AF89D2', '#51CDBA'],
+                colorId: $scope.currentSubjectId,
+                isMobile: false,
+                height: optionsPerDevice.height,
+                type: 'multi',
+                isMax: true,
+                max: 29,
+                min: 0,
+                subPoint: 35,
+                distance: optionsPerDevice.distance,
+                lineWidth: 2,
+                images: TimelineSrv.getImages(),
+                numbers: {
+                    font: '200 12px Lato',
+                    fillStyle: '#4a4a4a'
+                },
+                onFinish: onFinishCb
+            }
+        }
+
+        if (!content) {
+            content = '<canvas znk-timeline timeline-data="timeLineData" timeline-settings="options"></canvas>'
+        }
+
+        content = $compile(content)($scope);
+        $scope.$digest();
+
+        return {
+            scope: $scope,
+            content: content
+        };
+    }
+
+    it('should have retina support (width and height * 2 from the style width and height)', function () {
+        var scopeContent = createDirectiveHtml();
+        var content = scopeContent.content[0];
+        expect(content.width).toBeDefined();
+        expect(content.height).toEqual(300);
+        expect(content.style.width).toBeDefined();
```

Comment:
the test not render the width the same in phantom as in browser, so im just checking there are width and width style on the canvas element